curriculum/challenges/english/25-front-end-development/quiz-bash-and-sql/66f1affc0ef4fcca423d4688.md

```diff
@@ -21,435 +21,501 @@ Placeholder question
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You are working on a Linux system and need to write a bash script that performs these tasks:
+1. List all files and folders, including hidden ones, within the "/home/user" directory.
+2. Copy all ".txt" files from the "/home/user" directory to a folder named "text_backup" within the current directory.
+3. Delete all files with the ".tmp" extension in the "/home/user" directory, asking for confirmation before deleting each file.
 
 #### --distractors--
 
-Placeholder distractor 1
+```
+ls -a /home/user
+cp *.txt text_backup/
+rm -i *.tmp
+```
 
 ---
 
-Placeholder distractor 2
+```
+ls -a /home/user
+cp *.txt /home/user/text_backup/
+rm -rf /home/user/*.tmp
+```
 
 ---
 
-Placeholder distractor 3
+```
+ls -a /home/user
+cp /home/user/*.txt /home/user/text_backup/
+rm -i *.tmp
+```
+
 
 #### --answer--
 
-Placeholder answer
+```
+ls -a /home/user
+cp /home/user/*.txt text_backup/
+rm -i /home/user/*.tmp 
+```
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You are designing a database for a music store. You need to keep track of albums and their artists. You have two tables:
+
+**Table: Albums**
+
+| Column Name | Data Type  | Constraints |
+|-------------|------------|-------------|
+| AlbumID     | INT        | PRIMARY KEY |
+| AlbumTitle   | VARCHAR(255) | NOT NULL   |
+| ReleaseYear  | INT        |             |
+
+**Table: Artists**
+
+| Column Name | Data Type  | Constraints |
+|-------------|------------|-------------|
+| ArtistID    | INT        | PRIMARY KEY |
+| ArtistName  | VARCHAR(255) | NOT NULL   |
+
+You need to ensure that each album is associated with the correct artist.  
+
+**Question:** What is the best way to represent the relationship between the Albums table and Artists table, given that an artist can have multiple albums? 
+
 
 #### --distractors--
 
-Placeholder distractor 1
+Add a column named `ArtistName` to the Albums table.
 
 ---
 
-Placeholder distractor 2
+Add a column named `AlbumTitle` to the Artists table.
 
 ---
 
-Placeholder distractor 3
+Create a new table called `AlbumsArtists` with both `AlbumID` and `ArtistID`  as foreign keys.
 
 #### --answer--
 
-Placeholder answer
+Add a column named `ArtistID` to the Albums table as a foreign key referencing the `ArtistID` in the Artists table.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+You have been given a task to create a database that can hold information of customers and their addresses for some company. You would assume that you are to record all the details of the customers and should let a customer hold multiple addresses.
+
+Consider the following two tables:
+
+**Table: Customers**
+
+| Column Name | Data Type    | Constraints |
+| :---------- | :----------- | :---------- |
+| CustomerID  | INT          | PRIMARY KEY |
+| FirstName   | VARCHAR(255) | NOT NULL   |
+| LastName    | VARCHAR(255) | NOT NULL   |
+| Email       | VARCHAR(255) |             |
```

Comment:
````suggestion
```md
| Column Name | Data Type    | Constraints |
| :---------- | :----------- | :---------- |
| CustomerID  | INT          | PRIMARY KEY |
| FirstName   | VARCHAR(255) | NOT NULL   |
| LastName    | VARCHAR(255) | NOT NULL   |
| Email       | VARCHAR(255) |             |
```
````